lib/simp/rake/pupmod/helpers.rb

```diff
@@ -146,6 +146,98 @@ def define_tasks
       puts changelog[module_version]
     end
 
+    desc <<-EOM
+    Compare to latest tag.
+      ARGS:
+        * :tags_source => Set to the remote from which the tags for this
+                      project can be fetched, e.g. 'upstream' for a
+                      forked project. Defaults to 'origin'.
+        * :verbose => Set to 'true' if you want to see detailed messages
+
+      NOTES:
+      Compares mission-impacting (significant) files with the latest
+      tag and identifies the relevant files that have changed.  
+
+      Does nothing if the project owner, as specified in the
+      metadata.json file, is not 'simp'.
+
+      When mission-impacting files have changed, fails if
+      (1) Latest version cannot be extracted from the top-most
+          CHANGELOG entry.
+      (2) The latest version in the CHANGELOG (minus the release
+          qualifier) does not match the version in the metadata.json
+          file.
+      (3) A version bump is required but not recorded in both the
+          CHANGELOG and metadata.json files.
+      (4) The latest version is < latest tag.
+
+      Changes to the following files/directories are not considered
+      significant:
+      - Any hidden file/directory (entry that begins with a '.')
+      - Gemfile
+      - Gemfile.lock
+      - Rakefile
+      - spec directory
+      - doc directory
+    EOM
+    task :compare_latest_tag, [:tags_source, :verbose] do |t,args|
+      require 'json'
+      require 'puppet/util/package'
+
+      tags_source = args[:tags_source].nil? ? 'origin' : args[:tags_source]
+      verbose = true if args[:verbose].to_s == 'true' 
+
+      metadata = JSON.load(File.read('metadata.json'))
+      module_version = metadata['version']
+      owner =  metadata['name'].split('-')[0]
+
+      if owner == 'simp'
+        # determine last tag
+        `git fetch -t #{tags_source} 2>/dev/null`
+        tags = `git tag -l`.split("\n")
```

Comment:
If we assume that, in the future, all important tags will be annotated tags, then this might be easier:

`git for-each-ref --sort=taggerdate --format '%(refname) %(taggerdate)' refs/tags`

This could be added as an enhancement ticket if you agree.